algorithms/src/crh/bowe_hopwood_pedersen.rs

```diff
@@ -106,30 +110,21 @@ impl<G: Group, S: PedersenSize> CRH for BoweHopwoodPedersenCRH<G, S> {
                 S::NUM_WINDOWS,
             ));
         }
-
-        // Pad the input if it is not the current length.
-        let mut input_bytes = input;
-        let mut padded_input_bytes = vec![];
-        if (input.len() * 8) < S::WINDOW_SIZE * S::NUM_WINDOWS {
-            padded_input_bytes.extend_from_slice(input_bytes);
-            padded_input_bytes.resize((S::WINDOW_SIZE * S::NUM_WINDOWS) / 8, 0u8);
-            input_bytes = padded_input_bytes.as_slice();
+        // we cant use these in array sizes since they are from a trait (and cant be refered to at const time)
+        assert!(S::WINDOW_SIZE <= 256);
+        assert!(S::NUM_WINDOWS <= 296);
```

Comment:
Are these bounds from the declaration in `dpc/instantiated.rs`? These sizes are not entirely fixed and may be changed if we decide to update the proof systems or alter the circuits (which will lead to larger/smaller window requirements).

Reply:
Yeah, the intention is to change these when necessary. It's annoying, but this area is so hot it's worth it for the performance gains.

Reply:
FWIW, I tried using const generics on the struct instead, same obscure rust shenanigans about psuedo consts.

Reply:
In any case, this should be technically correct, however not sure if hard coding a bound here is the correct approach.

Reply:
For now, I think it's ok. But we may want to add more explicit documentation about why these bounds were selected and the cases for which they would need to be updated.

Reply:
See my other comment for a placeholder proposal for the time being